app/ui.js

```diff
@@ -321,6 +321,8 @@ const UI = {
     addClipboardHandlers() {
         document.getElementById("noVNC_clipboard_button")
             .addEventListener('click', UI.toggleClipboardPanel);
+        document.getElementById("noVNC_clipboard_text")
+            .addEventListener('input', UI.syncClipboardPanelToLocalClipboard);
```

Comment:
Does the 'input' event trigger when text is removed?

Shouldn't we use the same event as below, 'change'?

Reply:
The 'input' event even works when text is removed, e.g it fires on every text change immediately, instead of the 'change' event, which reacts only when you leave the text field. 

For seameless UX, my approach send the content from localClipboard via RFB when the vnc canvas event 'focus' is triggered. (e.g. it reads the content from localClipboard when clicking the browser canvas for example after copying new content). If somebody changes the content of clipboardPanel, the content additionally has to be written in localClipboard that on 'focus' the correct content is sent. When using only the 'change' event it would copy the content from the clipboardPanel to the localClipboard only after the 'focus' event was triggered (results to the wrong content is copied). That's why I use the 'input' event to synchronize the content from the clipboardPanel to the localClipboard first, before 'focus' is triggered.

Reply:
I see, thank you for a good explanation.